x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/type/DataTypeConversion.java

```diff
@@ -170,6 +188,8 @@ private static Conversion conversion(DataType from, DataType to) {
                 return conversionToDouble(from);
             case DATE:
                 return conversionToDate(from);
+            case TIME:
```

Comment:
Curious - what's with the order? Is not alphabetical, it's not priority base (datetime occurs more often than date or time).

Reply:
The logic was DATE (only), TIME (only), DATETIME (both)
Will change the ordering everywhere to be by priority: DATETIME, DATE, TIME
but if you don't mind I'll do it in a followup PR.